### PR TITLE
Create article and assign it to a category

### PR DESCRIPTION
**Pull request by Noelryn**
## PT LInk
https://www.pivotaltracker.com/story/show/161308257

## Included in PR

- Created category model
- Added a reference to articles model
- Added a foreign key to the article schema
- Added categories to seed file 
- Added dropdown menu to article creation form for selecting category

## What we learned 

- Creating a model 
- Adding a reference to the controller
- Using seed files

## Screenshot
screenshot 2018-10-22 at 20 59 16

### CODE REVIEW

**Review by Prozak8 — COMMENTED** [review]
Good testing and great naming

File: app/models/article.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class Article < ApplicationRecord
-  validates_presence_of :title, :content, :author
+  belongs_to :category
+  validates_presence_of :title, :content, :author, :category_id
```
Comment (tochman) [inline app/models/article.rb]: Don't think you have to add `:category_id` A belongs_to`association requires presence by default.

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -13,3 +13,14 @@
 Given("I click {string}") do |button|
   click_button button
 end
+
+Given("I select {string} from {string}") do |category, dropdown|
+  select category, from: dropdown
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Will this step always be used for selecting categories. I think you could reuse it on other dropdowns (as you call them) as well. I would rename the variable.

**Review by tochman — APPROVED** [review]
Great job. Small comments. Please address them and have your peers merge it.

File: app/models/article.rb (hunk 1 above)
Comment (Noelryn) [reply app/models/article.rb]: Ok great I'll change that

Comment (Noelryn) [reply app/models/article.rb]: I removed category_id from the article model, all test are still passing ...nice

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (Noelryn) [reply features/step_definitions/basic_steps.rb]: Thanks for the suggestion, I know you hate our naming. I am going to bring it up with the team in the morning about changing it to something better.

Comment (tochman) [reply features/step_definitions/basic_steps.rb]: I don't hate it @Noelryn, I just think you can abstract better.

**Pull request merged**